great_expectations/dataset/sqlalchemy_dataset.py

```diff
@@ -882,6 +886,70 @@ def column_reflection_fallback(self):
             ]
         return col_dict
 
+    ###
+    ###
+    ###
+    #
+    # Column Aggregate Expectation Implementations
+    #
+    ###
+    ###
+    ###
+
+    # noinspection PyUnusedLocal
+    @DocInherit
+    @MetaSqlAlchemyDataset.expectation(["other_table_name"])
+    def expect_table_row_count_to_equal_other_table(
+        self,
+        other_table_name,
+        result_format=None,
+        include_config=True,
+        catch_exceptions=None,
+        meta=None,
+    ):
+        """Expect the number of rows in this table to equal the number of rows in a different table.
+
+        expect_table_row_count_to_equal is a :func:`expectation \
+        <great_expectations.data_asset.data_asset.DataAsset.expectation>`, not a
+        ``column_map_expectation`` or ``column_aggregate_expectation``.
+
+        Args:
+            other_table_name (str): \
+                The name of the other table to which to compare.
+
+        Other Parameters:
+            result_format (string or None): \
+                Which output mode to use: `BOOLEAN_ONLY`, `BASIC`, `COMPLETE`, or `SUMMARY`.
+                For more detail, see :ref:`result_format <result_format>`.
+            include_config (boolean): \
+                If True, then include the expectation config as part of the result object. \
+                For more detail, see :ref:`include_config`.
+            catch_exceptions (boolean or None): \
+                If True, then catch exceptions and include them as part of the result object. \
+                For more detail, see :ref:`catch_exceptions`.
+            meta (dict or None): \
+                A JSON-serializable dictionary (nesting allowed) that will be included in the output without \
+                modification. For more detail, see :ref:`meta`.
+
+        Returns:
+            A JSON-serializable expectation result object.
```

Comment:
Should we search and replace these to ExpectationSuiteValidationResult objects?

Reply:
Yes!

Reply:
Updated.